util/movement-types/src/block.rs

```diff
@@ -106,6 +106,23 @@ impl Block {
 	pub fn add_transaction(&mut self, transaction: Transaction) {
 		self.transactions.insert(transaction);
 	}
+
+	pub fn collapse(blocks: Vec<Block>) -> Block {
+		let mut transactions = BTreeSet::new();
+		let parent = if let Some(first_block) = blocks.first() {
```

Comment:
The field is essentially bogus since we don't initialize the chain parent correctly on DA sequencer restart. But within movement-types this is the right thing to do.

Reply:
All this function is as an example to explain the idea of batch aggregation.